### PR TITLE
Block all routes except Home when user isn't logged

### PR DESCRIPTION
**Pull request by corneadoug**
### What is this PR for?
When user isn't logged while the auth is activated, we are hiding the notebooks and most of the menus.
However you can still access them using the url routes.
This PR is blocking the url routes except the home route ('/') in that case.

### What type of PR is it?
Improvement

### What is the Jira issue?
https://issues.apache.org/jira/browse/ZEPPELIN-1123

### How should this be tested?
* Activate the auth in shiro.ini
* Launch Zeppelin, and try to visit `localhost:8080/#/interpreter` by typing it in your browser
* You should be redirected to '/'

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

File: zeppelin-web/src/app/app.js [hunk 1]
```diff
@@ -109,11 +108,18 @@
   }
 
   function bootstrapApplication() {
+    zeppelinWebApp.run(function($rootScope, $location) {
+      $rootScope.$on('$routeChangeStart', function(event, next, current) {
+        if (!$rootScope.ticket && !next.$$route.publicAccess) {
+          event.preventDefault();
```
Comment (prabhjyotsingh) [inline zeppelin-web/src/app/app.js]: `event.preventDefault();` is not letting my page to render. On removing this works as expected.

**Comment by prabhjyotsingh** [comment]
LGTM! minor suggestion.

**Comment by prabhjyotsingh** [comment]
Tried new changes works well.

**Comment by corneadoug** [comment]
CI doesn't seem related, but still gonna retrigger it

**Comment by corneadoug** [comment]
Ci is green, merging if there is no more discussions

**Pull request closed without merging**